File: typescript-client-sdk/src/sdk/models/operations/getdatacompaniescompanyidconnectionsconnectionidassessfinancialmetrics.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import * as shared from "../shared";
import { Transform, Type } from "class-transformer";


export class GetDataCompaniesCompanyIdConnectionsConnectionIdAssessFinancialMetricsPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=companyId" })
  companyId: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=connectionId" })
  connectionId: string;
}

export class GetDataCompaniesCompanyIdConnectionsConnectionIdAssessFinancialMetricsQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=numberOfPeriods" })
  numberOfPeriods: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=periodLength" })
  periodLength: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=reportDate;dateTimeFormat=YYYY-MM-DDThh:mm:ss.sssZ" })
  reportDate: Date;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=showMetricInputs" })
  showMetricInputs?: boolean;
}

export class GetDataCompaniesCompanyIdConnectionsConnectionIdAssessFinancialMetricsSecurity extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, scheme=true;type=apiKey;subtype=header" })
  apiKeyAuth?: shared.SchemeAPIKeyAuth;

  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  codatLogin?: shared.SchemeCodatLogin;
}

export class GetDataCompaniesCompanyIdConnectionsConnectionIdAssessFinancialMetricsRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: GetDataCompaniesCompanyIdConnectionsConnectionIdAssessFinancialMetricsPathParams;

  @SpeakeasyMetadata()
  queryParams: GetDataCompaniesCompanyIdConnectionsConnectionIdAssessFinancialMetricsQueryParams;

  @SpeakeasyMetadata()
  security: GetDataCompaniesCompanyIdConnectionsConnectionIdAssessFinancialMetricsSecurity;
}

export class GetDataCompaniesCompanyIdConnectionsConnectionIdAssessFinancialMetricsResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  codatAssessDataContractsFinancialsMetricsFinancialMetricsDataSet?: shared.CodatAssessDataContractsFinancialsMetricsFinancialMetricsDataSet;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}

File: go-client-sdk/pkg/models/operations/getdatacompaniescompanyidconnectionsconnectionidassessfinancialmetrics.go
package operations

import (
	"github.com/speakeasy-sdks/codat-sdks/go-client-sdk/v2/pkg/models/shared"
	"time"
)

type GetDataCompaniesCompanyIDConnectionsConnectionIDAssessFinancialMetricsPathParams struct {
	CompanyID    string `pathParam:"style=simple,explode=false,name=companyId"`
	ConnectionID string `pathParam:"style=simple,explode=false,name=connectionId"`
}

type GetDataCompaniesCompanyIDConnectionsConnectionIDAssessFinancialMetricsQueryParams struct {
	NumberOfPeriods  int       `queryParam:"style=form,explode=true,name=numberOfPeriods"`
	PeriodLength     int       `queryParam:"style=form,explode=true,name=periodLength"`
	ReportDate       time.Time `queryParam:"style=form,explode=true,name=reportDate"`
	ShowMetricInputs *bool     `queryParam:"style=form,explode=true,name=showMetricInputs"`
}

type GetDataCompaniesCompanyIDConnectionsConnectionIDAssessFinancialMetricsSecurity struct {
	APIKeyAuth *shared.SchemeAPIKeyAuth `security:"scheme,type=apiKey,subtype=header"`
	CodatLogin *shared.SchemeCodatLogin `security:"scheme,type=oauth2"`
}

type GetDataCompaniesCompanyIDConnectionsConnectionIDAssessFinancialMetricsRequest struct {
	PathParams  GetDataCompaniesCompanyIDConnectionsConnectionIDAssessFinancialMetricsPathParams
	QueryParams GetDataCompaniesCompanyIDConnectionsConnectionIDAssessFinancialMetricsQueryParams
	Security    GetDataCompaniesCompanyIDConnectionsConnectionIDAssessFinancialMetricsSecurity
}

type GetDataCompaniesCompanyIDConnectionsConnectionIDAssessFinancialMetricsResponse struct {
	CodatAssessDataContractsFinancialsMetricsFinancialMetricsDataSet *shared.CodatAssessDataContractsFinancialsMetricsFinancialMetricsDataSet
	ContentType                                                      string
	StatusCode                                                       int
}
